utils/docker/build.sh

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env bash
+#
+# Copyright 2017-2019, Intel Corporation
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#
+#     * Redistributions of source code must retain the above copyright
+#       notice, this list of conditions and the following disclaimer.
+#
+#     * Redistributions in binary form must reproduce the above copyright
+#       notice, this list of conditions and the following disclaimer in
+#       the documentation and/or other materials provided with the
+#       distribution.
+#
+#     * Neither the name of the copyright holder nor the names of its
+#       contributors may be used to endorse or promote products derived
+#       from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#
+# build.sh - runs a Docker container from a Docker image with environment
+#            prepared for running pmemkv-ruby build and tests.
+#
+#
+# Notes:
+# - run this script from its location or set the variable 'HOST_WORKDIR' to
+#   where the root of this project is on the host machine,
+# - set variables 'OS' and 'OS_VER' properly to a system you want to build this
+#   repo on (for proper values take a look on the list of Dockerfiles at the
+#   utils/docker/images directory), eg. OS=ubuntu, OS_VER=16.04.
+#
+
+set -e
+
+if [[ -z "$OS" || -z "$OS_VER" ]]; then
+	echo "ERROR: The variables OS and OS_VER have to be set " \
+		"(eg. OS=fedora, OS_VER=30)."
+	exit 1
+fi
+
+if [[ -z "$HOST_WORKDIR" ]]; then
+	HOST_WORKDIR=$(readlink -f ../..)
+fi
+
+chmod -R a+w $HOST_WORKDIR
+
+if [[ "$TRAVIS_EVENT_TYPE" == "cron" || "$TRAVIS_BRANCH" == "coverity_scan" ]]; then
+	if [[ "$TYPE" != "coverity" ]]; then
+		echo "Skipping non-Coverity job for cron/Coverity build"
+		exit 0
+	fi
+else
+	if [[ "$TYPE" == "coverity" ]]; then
+		echo "Skipping Coverity job for non cron/Coverity build"
+		exit 0
+	fi
+fi
+
+imageName=${DOCKERHUB_REPO}:${OS}-${OS_VER}
+containerName=pmemkv-ruby-${OS}-${OS_VER}
+
+if [[ "$command" == "" ]]; then
+	case $TYPE in
+		pmemkv-master)
+			command="./run-build.sh master";
```

Comment:
Testing a moving target (branch name) will lead to random failures unrelated to changed in pmemkv-ruby. Please use a commit id.

This means that we can build required versions of pmemkv at the Docker image rebuild time.